### DISCUSSION TITLE
隐语TEEU仿真执行没有反应

### DISCUSSION THREAD

**Discussion by jie-lab** [secretflow/secretflow#778] [Q&A] [answered]
隐语TEEU仿真：occlum run /bin/python /root/demo.py 执行此指令，demo没有反应，与bob和alice的通信连接不上，利用occlum编译的hello_world.c文件，可以编译成功，但是也是无法执行的。
仿真环境的搭建条件是：
利用docker镜像直接拉取搭建的
bob和alice仿真通信是可以的

**Comment by zhouaihui** [comment]
hi， @jie-lab ，可以提供一下你执行的所有步骤的命令吗？

**Comment by jie-lab** [comment]
你好，这个问题产生的原因已经知道了，因为内存不足造成的，另一问题是这个teeu仿真为啥需要这么大的内存呢。

**Comment by zhouaihui** [comment, marked as answer]
目前基于SGX2的occlum还不支持虚拟内存，occlum其实是单地址空间的OS，所有申请的内存都会被认为是实际占用。而运行SecretFlow需要依赖很多基础组件（比如ray、grpc、numpy等），这些软件运行时会申请大量内存，在支持虚拟内存的OS（比如linux）上时并不会有问题（因为只有实际使用到的时候才会分配物理内存），但在occlum上运行时的表现会变成占用很大内存。

**Comment by jie-lab** [comment]
嗯，谢谢周老师指点。

**Comment by zhouaihui** [comment]
欢迎下次从issue提问。github的discussion有点问题，无法推送到我们的robot，容易遗漏，issue可以正常推送，会有值班人员跟进。
